internal/p2p/channel.go

```diff
@@ -259,17 +264,85 @@ func (ch *libp2pChannelImpl) canonicalizedTopicName() string {
 }
 
 func (ch *libp2pChannelImpl) Receive(ctx context.Context) *ChannelIterator {
+	// TODO: consider caching an iterator in the channel, or
+	// erroring if this gets called more than once.
+	//
+	// While it's safe to register a handler more than once, we
+	// could get into a dodgy situation where if you call receive
+	// more  than once, the subsequently messages won't be routed
+	// correctly.
 	iter := &ChannelIterator{
 		pipe: make(chan Envelope),
 	}
 
+	ch.host.SetStreamHandler(protocol.ID(ch.canonicalizedTopicName()), func(stream network.Stream) {
+		buf := bufio.NewReader(stream)
+		// TODO: change the max delimited message into
+		// something that's passed in as a parmeter.
+		reader := protoio.NewDelimitedReader(buf, 1023)
```

Comment:
It doesn't seem like we should be setting this small a limit. This seems more like a safety measure ("node doesn't explode") than a resource knob.

Given that we ship transactions around, we probably ought to have this at least as large as the max transaction size (and probably a multiple of that, just so this doesn't happen unless something really odd is going down).

Reply:
ah yea, it's historically configurable at the top level, and I was hoping to plumb in the current value, but it's a bit hard to get the config from there to here, and just had 1024 (or 1023, I appear to have missed by one) as a placeholder for the moment.

Reply:
My sense is that we shouldn't have this be a knob at all, but that's a broader question we probably don't need to answer yet.